### ISSUE TITLE
TodoList Tool is not accessible from the keyboard

### ISSUE THREAD

**Issue by accesswatch** [microsoft/vscode#260833]
Type: <b>Bug</b>

1. Open VSCode and ask a prompt that would start planning and use a to do list such as:

Help me develop a fully accessible Monopoly game that is screen reader friendly and use a todo list.

2. The todo list will initially be shown in chat and is visible with alt+f2.

3. Shift + Tab from the Chat input field until you reach the Tasks button and it will show its state as expanded. I should be able to expand this with the arrow keys I think but space and enter will perform these actions.

4. Using the Alt + F2 Hotkey yeilds the output from chat and not the task list.

Additional Notes:

This comes from the release notes from the latest version of VSCode:

Track progress with task lists

Setting: chat.todoListTool.enabled

The great thing about agent mode is that you can give it a high-level task and have it implement it. As it plans the work and breaks it down into smaller tasks, it can be overwhelming to track the progress of all these individual tasks.

We are introducing the task/todo list feature in chat to better help you see which tasks are completed and which ones are still pending. You can view the task list at the top of the Chat view, so you always have visibility into the progress being made. As the agent progresses through its work, it updates the task list.

Expectations:

I would love for this to do list to remain visible and accessible in view some how. This is a cool feature to watch the results of vibing in VSCode. In addition, the checked state when reviewing the window with the JAWS cursor does not show the checked or unchecked state of items. The JAWS cursor is not a solution. Native support should be made available to be accessible.

VS Code version: Code 1.103.0 (e3550cfac4b63ca4eafca7b601f0d2885817fd1f, 2025-08-06T21:40:10.271Z)
OS version: Windows_NT x64 10.0.26100
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 5 Microsoft Surface (R) Edition (12 x 2196)|
|GPU Status|2d_canvas: enabled<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>trees_in_viz: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled<br>webnn: disabled_off|
|Load (avg)|undefined|
|Memory (System)|7.45GB (0.61GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (78)</summary>

Extension|Author (truncated)|Version
---|---|---
jelly-language-support|Act|1.2.2
Bookmarks|ale|13.5.0
mcp-store|Aut|1.0.0
vscode-intelephense-client|bme|1.14.4
vscode-innosetup|cho|1.9.3
vscode-pandoc|Chr|0.6.2
promptboost|chr|0.8.0
npm-intellisense|chr|1.4.5
github-copilot-code-review|Cod|0.0.4
vscode-markdownlint|Dav|0.60.0
vscode-eslint|dba|3.0.16
vscode-faker|dee|2.0.0
docker|doc|0.14.0
githistory|don|0.6.20
xml|Dot|2.5.1
prettier-vscode|esb|11.0.0
vscode-firefox-debug|fir|2.15.0
macros|ged|1.2.1
codespaces|Git|1.17.3
copilot|Git|1.350.0
copilot-chat|Git|0.30.0
remotehub|Git|0.64.0
vscode-github-actions|git|0.27.2
vscode-pull-request-github|Git|0.116.0
vscode-settings-cycler|hoo|1.0.1
vscode-htmlhint|HTM|1.11.1
accessibility-alert-toggle|Jef|0.0.1
copy-markdown-as-html|jer|1.1.0
mark-sharp|jon|1.8.1
language-matlab|Mat|1.3.4
markdown-shortcuts|mdi|0.12.0
rainbow-csv|mec|3.20.0
copilot-code-review|mka|1.0.6
vscode-azure-mcp-server|ms-|0.5.4
vscode-containers|ms-|2.1.0
vscode-docker|ms-|2.0.0
csdevkit|ms-|1.41.11
csharp|ms-|2.87.31
vscode-dotnet-runtime|ms-|2.3.7
vscode-edge-devtools|ms-|2.1.9
datawrangler|ms-|1.22.0
remote-containers|ms-|0.424.0
remote-ssh|ms-|0.121.2025073115
remote-ssh-edit|ms-|0.87.0
remote-wsl|ms-|0.99.0
vscode-remote-extensionpack|ms-|0.26.0
live-server|ms-|0.4.15
remote-explorer|ms-|0.6.2025072209
remote-repositories|ms-|0.43.2025072801
remote-server|ms-|1.6.2025072309
vscode-speech|ms-|0.16.0
vscode-typescript-next|ms-|6.0.20250809
vscode-websearchforcopilot|ms-|0.1.2
vsliveshare|ms-|1.0.5948
windows-ai-studio|ms-|0.18.3
html-snippets-pro|rit|1.0.1
file-bunny|rob|2.4.0
markdown-shortcuts|rob|0.7.1
markdown-snippets|rob|0.9.1
marky-dynamic|rob|0.4.2
marky-edit|rob|0.7.0
marky-markdown|rob|2.10.0
marky-stats|rob|0.11.0
snippets-ranger|rob|1.8.0
joplin-vscode-plugin|rxl|1.7.3
partial-diff|ryu|1.4.3
markdown-preview-enhanced|shd|0.8.18
code-spell-checker|str|4.0.47
wav-preview|suk|2.6.0
markdown-live-preview|Tec|1.5.0
intellicode-api-usage-examples|Vis|0.2.9
vscodeintellicode|Vis|1.3.2
gitdoc|vsl|0.2.3
wordpress-toolbox|wor|1.3.16
fake-data|wyc|1.0.0
php-debug|xde|1.37.0
local-history|xyz|1.8.1
markdown-all-in-one|yzh|3.6.3

</details>

**Comment by taylorarndt** [comment]
I am having the same issue. I am using NVDA.

**Comment by accesswatch** [comment]
Is this something that would be worthy of a fix @meganrogge ?

**Comment by meganrogge** [comment]
Yes we need to fix these issues

**Comment by meganrogge** [comment]
I agree. I had this TPI and created some accessibility issues from it - definitely needs work still.

**Comment by meganrogge** [comment]
It would be good if you could do so pls .
